rmf_site_editor/src/widgets/view_model_instances.rs

```diff
@@ -0,0 +1,241 @@
+#[derive(Default)]
+pub struct ViewModelInstancesPlugin {}
+
+impl Plugin for ViewModelInstancesPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_plugins(PropertiesTilePlugin::<ViewModelInstances>::new());
+    }
+}
+
+#[derive(SystemParam)]
+pub struct ViewModelInstances<'w, 's> {
+    scenarios: Query<
+        'w,
+        's,
+        (Entity, &'static NameInSite, &'static mut Scenario<Entity>),
+        With<ScenarioMarker>,
+    >,
+    change_current_scenario: EventWriter<'w, ChangeCurrentScenario>,
+    current_scenario: ResMut<'w, CurrentScenario>,
+    model_descriptions: Query<
+        'w,
+        's,
+        (Entity, &'static NameInSite, Option<&'static SiteID>),
+        (With<ModelMarker>, With<Group>),
+    >,
+    model_instances: Query<
+        'w,
+        's,
+        (
+            Entity,
+            &'static NameInSite,
+            &'static Category,
+            &'static Affiliation<Entity>,
+            Option<&'static SiteID>,
+        ),
+        With<InstanceMarker>,
+    >,
+    selection: Res<'w, Selection>,
+    select: EventWriter<'w, Select>,
+    delete: EventWriter<'w, Delete>,
+    reset_pose: EventWriter<'w, ResetPose>,
+}
+
+impl<'w, 's> WidgetSystem<Tile> for ViewModelInstances<'w, 's> {
+    fn show(_: Tile, ui: &mut Ui, state: &mut SystemState<Self>, world: &mut World) -> () {
+        let mut params = state.get_mut(world);
+        CollapsingHeader::new("Models")
+            .default_open(true)
+            .show(ui, |ui| {
+                params.show_widget(ui);
+            });
+    }
+}
+
+impl<'w, 's> ViewModelInstances<'w, 's> {
+    pub fn show_widget(&mut self, ui: &mut Ui) {
+        if let Some(current_scenario_entity) = self.current_scenario.0 {
+            if let Ok((_, _, mut scenario)) = self.scenarios.get_mut(current_scenario_entity) {
+                let current_scenario_instances = scenario.instances.clone();
+                let mut non_affiliated_instances = HashSet::<Entity>::new();
+                ScrollArea::vertical()
+                    .max_height(INSTANCES_VIEWER_HEIGHT)
+                    .auto_shrink([false, false])
+                    .show(ui, |ui| {
+                        for (desc_entity, desc_name, _) in self.model_descriptions.iter() {
+                            CollapsingHeader::new(desc_name.0.clone())
+                                .id_source(desc_name.0.clone())
+                                // TODO(@xiyuoh) true if model is selected
+                                .default_open(false)
+                                .show(ui, |ui| {
+                                    for (
+                                        instance_entity,
+                                        instance_name,
+                                        category,
+                                        affiliation,
+                                        instance_site_id,
+                                    ) in self.model_instances.iter_mut()
```

Comment:
I'm noticing instability in the ordering of this when I select or deselect instances.

I suspect a more efficient and more stable way of doing this iteration would be to iterate over the [`Members`](https://github.com/open-rmf/rmf_site/blob/a9ac2a4ad03903cf1b96730238b9bfddd430077e/rmf_site_editor/src/site/group.rs#L31) of the description group.